src/common/types/row_data_collection.cpp

```diff
@@ -46,12 +55,12 @@ void RowDataCollection::TemplatedSerializeVectorSortable(VectorData &vdata, cons
 	} else {
 		for (idx_t i = 0; i < add_count; i++) {
 			auto idx = sel.get_index(i);
-			auto source_idx = vdata.sel->get_index(idx);
```

Comment:
The branch with `has_null = false` appears to be untested - could you add a test with a list that involves no null values?

Reply:
This was due to not using ValidityStatistics while serializing. This branch should now be used again. Waiting for CI